api/src/main/java/org/openmrs/module/kenyaemr/calculation/library/hiv/LostToFollowUpCalculation.java

```diff
@@ -43,47 +56,57 @@ public String getFlagMessage() {
 	 */
 	@Override
 	public CalculationResultMap evaluate(Collection<Integer> cohort, Map<String, Object> arg1, PatientCalculationContext context) {
 
 		Program hivProgram = MetadataUtils.existing(Program.class, HivMetadata._Program.HIV);
 		Concept reasonForDiscontinuation = Dictionary.getConcept(Dictionary.REASON_FOR_PROGRAM_DISCONTINUATION);
 		Concept transferout = Dictionary.getConcept(Dictionary.TRANSFERRED_OUT);
-
 		Set<Integer> alive = Filters.alive(cohort, context);
 		Set<Integer> inHivProgram = Filters.inProgram(hivProgram, alive, context);
 
-		//CalculationResultMap lastEncounters = Calculations.lastEncounter(null, inHivProgram, context);
-		CalculationResultMap lastReturnDateObss = Calculations.lastObs(Dictionary.getConcept(Dictionary.RETURN_VISIT_DATE), inHivProgram, context);
 		CalculationResultMap lastProgramDiscontinuation = Calculations.lastObs(reasonForDiscontinuation, cohort, context);
 
 		CalculationResultMap ret = new CalculationResultMap();
 		for (Integer ptId : cohort) {
 			boolean lost = false;
+			Integer tcaConcept = 5096;
+			Date tcaDate = null;
 
-			// Is patient alive and in the HIV program
-			if (alive.contains(ptId)) {
+			// Is patient alive and in HIV program
+			if (inHivProgram.contains(ptId)) {
+				PatientService patientService = Context.getPatientService();
+				//With Greencard Encounter
+				EncounterType greenCardEncType = MetadataUtils.existing(EncounterType.class, HivMetadata._EncounterType.HIV_CONSULTATION);
+				Form pocHivFollowup = MetadataUtils.existing(Form.class, HivMetadata._Form.HIV_GREEN_CARD);
+				Form rdeHivFollowup = MetadataUtils.existing(Form.class, HivMetadata._Form.MOH_257_VISIT_SUMMARY);
+				Encounter lastFollowUpEncounter = EmrUtils.lastEncounter(patientService.getPatient(ptId), greenCardEncType, Arrays.asList(pocHivFollowup, rdeHivFollowup));  //last hiv followup encounter
 
-				// Patient is lost if no encounters in last X days
-				//Encounter lastEncounter = EmrCalculationUtils.encounterResultForPatient(lastEncounters, ptId);
-				Date lastScheduledReturnDate = EmrCalculationUtils.datetimeObsResultForPatient(lastReturnDateObss, ptId);
-				Obs discontuation = EmrCalculationUtils.obsResultForPatient(lastProgramDiscontinuation, ptId);
-				if (lastScheduledReturnDate != null) {
-					if(daysSince(lastScheduledReturnDate, context) > HivConstants.LOST_TO_FOLLOW_UP_THRESHOLD_DAYS){
-						lost = true;
-					}
-					if(discontuation != null && discontuation.getValueCoded().equals(transferout)) {
-						lost = false;
+				if (lastFollowUpEncounter != null) {
+					for (Obs obs : lastFollowUpEncounter.getObs()) {
+						if (obs.getConcept().getConceptId().equals(tcaConcept)) {
+							tcaDate = obs.getValueDatetime();
+							Obs discontinuation = EmrCalculationUtils.obsResultForPatient(lastProgramDiscontinuation, ptId);
+							if (tcaDate != null) {
+								if (daysSince(tcaDate, context) > HivConstants.LOST_TO_FOLLOW_UP_THRESHOLD_DAYS) {
```

Comment:
Either set threshold = 30 days and use > here or change this line to include 31 days  (>= )
like
 **if (daysSince(tcaDate, context) >= HivConstants.LOST_TO_FOLLOW_UP_THRESHOLD_DAYS)**